salt/crypt.py

```diff
@@ -520,9 +520,17 @@ def _authenticate(self):
                     error = SaltClientError('Detect mode is on')
                     break
                 if self.opts.get('caller'):
-                    print('Minion failed to authenticate with the master, '
-                          'has the minion key been accepted?')
-                    sys.exit(2)
+                    # We have a list of masters, so we should break
+                    # and try the next one in the list.
+                    if self.opts.get('local_masters', None):
```

Comment:
This is nit-picky but this is such a sensitive area that I would feel better if we checked explicitly against a list type.

Reply:
Oh! Oh! My fault. You're totally right.